### DISCUSSION TITLE
New to Gatsby, GraphQL Question + 404 Error

### DISCUSSION THREAD

**Discussion by ALPJeff** [gatsbyjs/gatsby#36703] [Help]
Good morning! 

I don't _think_ this is a duplicate question based on its simplicity alone, but I may have overlooked a similar discussion - I am only on my second cup of coffee. 

I've been thrown into the React/Gatsby world with my FT job & I enjoy the simplicity of it, despite the learning curve, and have been learning Gatsby + WP. There are different ways to accomplish this combo & I'm working on Jason Lengstorf's tutorial but when I hit step 5, I'm having an issue that I don't understand: 
https://egghead.io/lessons/gatsby-display-wordpress-page-content-in-gatsby-page-template-components

In Jason's tutorial, the graphql data displays immediately on local in this step but that's not the case with my code, which I've cloned into a repo here: 
https://github.com/ALPJeff/graphqlQuestion

I can see the gql  data if I access localhost:8000/___graphiql but otherwise, I'm seeing an automated 404 page & I don't know why. Am I missing something really obvious?

**Comment by imjoshin** [comment]
> I can see the gql data if I access localhost:8000/___graphiql but otherwise, I'm seeing an automated 404 page & I don't know why. Am I missing something really obvious?

Hey @ALPJeff !

First off, I see the `createPages` call is actually named `creatPages`, that would mean none of your blog post pages are actually being created. Secondly, the `gatsby-config` still points to `https://mycoolsite.com/graphql` for `gatsby-source-graphql`. I assume you have a different graphql host you intend to use?

Once these are resolved, I think you'll be all set. If not, here would be my thought process to debug further: I see there's a `src/pages/index.js` file. As it's named, it means that you'll be serving that page at `localhost:8000`. Based on the [gatsby-node](https://github.com/ALPJeff/graphqlQuestion/blob/main/gatsby-node.js) file, it also looks like you'll be serving files at `localhost:8000/${postUri}`. If those files aren't appearing, I'd add some log messages to `createPages` to see what blog posts are being processed and see if that aligns with your expectations.

**Reply by ALPJeff** [reply]
@j0sh77 - I thought I had corrected that typo on createPages so that's embarrassing on my part 😳 & yes, that URL is intentional on my part.
I updated that typo & removed that index.js file, which is also covered in Jason's course, but now I'm getting a new error to work on - an opportunity to excel 😂 

```
Your site's "gatsby-node.js" must set the page path when creating a page.

The page object passed to createPage:
{
    "path": null,
    "component":
"/GitHub/gatsby-starter-hello-world/src/templates/page-template.js",
    "context": {
        "id": "cG9zdDo0Nw=="
    }
}

See the documentation for the "createPage" action —
https://www.gatsbyjs.com/docs/reference/config-files/actions#createPage

not finished createPages - 1.070s
```

I understand that the id it's pulling back is one of the pages on my wp site but I don't get why I'm having this error - I ran clean beforehand, next I'll try starting all over, since this is barely into Jason's course & see if that resolves the issue.
